### PR TITLE
Feature/update manuf defn goats

### PR DESCRIPTION
**Pull request by jacobtani**
* Add new marking definition for 006 for participant code and mgmt number
* Add more failure handling 
* Fix manufacturing definition for allflex fdx leg bands
* Add new fixtures and fix specs

### CODE REVIEW

**Comment by jacobtani** [comment]
[(1) Configure Allflex mapper for 2 goat products](https://trello.com/c/1Flhym7p/2196-1-configure-allflex-mapper-for-2-goat-products)

File: lib/allflex/product_mapper/allflex_product.rb [hunk 1]
```diff
@@ -231,7 +238,8 @@ def marked?
       # the id will either be defined in the maufacturing definition,
       # or it will need to be build algorithmically from the manufacturing definition and the components
       def generate_id
-        return Failure("No ordered_id_components") unless ordered_id_components
+        return Failure(ordered_id_components: [{ code: 'No ordered product id components passed in', errors: { message: 'Error in generate_id method in allflex product' }, context: {} }]) unless ordered_id_components
```
Comment (josephleniston) [inline lib/allflex/product_mapper/allflex_product.rb]: Where is the test for this change?

```suggestion
        return Failure(ordered_id_components: [{ code: 'No ordered product id components passed in', errors: [{ message: 'Error in generate_id method in allflex product' }], context: {} }]) unless ordered_id_components
```

File: lib/allflex/product_mapper/product_builder.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@ def build(ordered_product, product_defn)
                                       components:         components.value_or,
                                       manufacturing_definition: product_defn
                                     )
-        product.id.success? ? Success(product) : Failure(product)
+        product.id.success? ? Success(product) : Failure(product_id_building: [{ code: 'Allflex Product Creation Error', errors: { message: 'Error in build method in Product Builder' }, context: { product: product } }])
```
Comment (josephleniston) [inline lib/allflex/product_mapper/product_builder.rb]: Where is the test for this change?

```suggestion
        product.id.success? ? Success(product) : Failure(product_id_building: [{ code: 'Allflex Product Creation Error', errors: [{ message: 'Error in build method in Product Builder' }], context: { product: product } }])
```

File: lib/allflex/product_mapper/allflex_product.rb (hunk 1 above)
Comment (jacobtani) [reply lib/allflex/product_mapper/allflex_product.rb]: Fixed the errors structure within the app too

Comment (jacobtani) [reply lib/allflex/product_mapper/allflex_product.rb]: Done a test for this one

File: lib/allflex/product_mapper/product_builder.rb (hunk 2 above)
Comment (jacobtani) [reply lib/allflex/product_mapper/product_builder.rb]: Haven't managed to be able to pull a test for this one sorry. Couldn't replicate it as it gets caught earlier and never goes to the Failure here from the specs we have.
Leave for now i think.

**Pull request merged**